portfolio/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -22,18 +24,28 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that displays login status of user. */
+/** Servlet that retrieves user login status. */
 @WebServlet("/user-login")
 public class UserServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html");
     UserService userService = UserServiceFactory.getUserService();
+    
+    Gson gson = new Gson();
     if (userService.isUserLoggedIn()) {
-      response.getWriter().println("<p>User logged in</p>");
+      String urlToRedirectToAfterUserLogsOut = "/html/comments.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      User newUser = User.create(/** loggedIn= */ true, /** loginURL= */ "", logoutUrl);
+      response.getWriter().println(gson.toJson(newUser));
     } else {
-      response.getWriter().println("<p>User not logged in</p>");
+      String urlToRedirectToAfterUserLogsIn = "/html/comments.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      User newUser = User.create(/** loggedIn= */ false, loginUrl, /** logoutURL= */ "");
```

Comment:
nit: for these comments it usually is /* */

Reply:
https://google.github.io/styleguide/javaguide.html#s4.8.6-comments shows you the types of comments which should be used. And then JavaDoc is something different: https://google.github.io/styleguide/javaguide.html#s7-javadoc

Reply:
done